client/script.js

```diff
@@ -0,0 +1,184 @@
+        const taskid = event.target.getAttribute('taskid');
+
+        fetch(`http://localhost:3000/task/${taskid}`, {
+          method: "PUT",
+          headers: {
+            "Content-Type": "application/json",
+            },
+            body: JSON.stringify({done: true})
+        }).then(res => {
+          if(res.ok) {
+            toggleCompleteTodo(taskid);
+			}
+        }).catch(error => {
+          console.log("do something with error")
+          });
+        });
+      });
+
+    const modifyButtons = todoItem.querySelectorAll('button.modifyButton');
+	  modifyButtons.forEach(button => {
+      button.addEventListener('click', modifyTodo);
+	 });
+	 
+	const deleteButtons = todoItem.querySelectorAll('button.deleteButton');
+    deleteButtons.forEach(button => {
+      button.addEventListener('click', function(event){
+        let taskid = event.target.getAttribute('taskid');
+        fetch(`http://localhost:3000/task/${taskid}`, {
+         method: "DELETE"
+         }).then(res => {
+          if (res.ok){
+		  deleteTodo(taskid);
+		  }
+          });
+         });
+        });
+  }
+	 
+	function toggleCompleteTodo(id) {
+  const taskElement = document.querySelector(`li[id='${id}']`);
+  if (taskElement.classList.contains("completed")) {
+    taskElement.classList.remove('completed');
+  } else {
+    taskElement.classList.add('completed');
+  }
+  
+}
+	
+	function deleteTodo(id){
+	const taskElement = document.querySelector(`li[id='${id}']`);
+	taskElement.remove();
+	}
+	
+	
+	function modifyTodo(event){
+	const button = event.target;
+	const taskid = button.getAttribute('taskid');
+	
+	const todoItem = button.parentNode.parentNode;
+	const todoTitle = todoItem.firstChild.textContent.trim();
+
+  const inputField = document.createElement('input');
+  inputField.type = 'text';
+  inputField.value = todoTitle;
+  const saveButton = document.createElement('button');
+  saveButton.textContent = 'Save';
+  saveButton.addEventListener("click", function (event) {
+    const newTitle = inputField.value.trim();
+    if (newTitle !== '') {
+      todoItem.firstChild.textContent = newTitle;
+      todoItem.removeChild(inputField);
+      todoItem.removeChild(saveButton);
+    } else {
+      alert('Please enter a valid title!');
+    }
+    fetch(`http://localhost:3000/todos/:${todoItem.id}`, {
+      method: "PATCH", 
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        id: +todoItem.id,
+        title: newTitle})
+      }).then(res => {
+          console.log(res.status);
+        })
+    });
+  
+    todoItem.appendChild(inputField);
+    todoItem.appendChild(saveButton);
+	}
+	
+	function removeCompletedTodos() {
+  const completedTodos = document.querySelectorAll('.completed');
+  completedTodos.forEach(todo => todo.remove());
+}
+
+
+document.querySelector('#newTodoContainer button:nth-of-type(1)').addEventListener('click', function () {

```

Comment:
try to avoid selectors like this one. 
it's extremely easy to break everything if you only rely on N-th position when programming behaviour. 
so, here you kinda think: lets my 1st button in this container does "this", and my second button is doing "that". 
next day, you find a bug in responsive design i get a request from UX designer asking you to make button container flex and able to be reversible, or add another button somewhere. and your N-th logic breaks. 
so instead, add an "Attribute" to the button. example: "name", or "class".